app/Models/Extensions/ToArrayThrowsNotImplemented.php

```
@@ -0,0 +1,19 @@
+<?php
+
+namespace App\Models\Extensions;
+
+use App\Exceptions\Internal\NotImplementedException;
+
+/**
+ * Trait ToArrayThrowsNotImplemented.
+ *
+ * Now that we use Resources toArray should no longer be used.
+ * Throw an exception if we encounter this function in the code.
+ */
+trait ToArrayThrowsNotImplemented
```

Is this really necessary? Calling a non-existing function should throw an exception anyways, and I don't see why we need something like this.

---

Unfortunately yes. The reason is that `Model` are of the type `Arrayable`, hence provide the method `toArray` by default. This method will convert the attributes except those mentioned in `$hidden`.
In theory, the method provided by this trait should not be reached in any code path, but I would like to really ensure this does not happen at all in a hidden array cast that we did not think of. :)